codegen/src/main/java/software/amazon/awssdk/codegen/poet/PoetUtils.java

```diff
@@ -109,4 +112,15 @@ public static JavaFile buildJavaFile(ClassSpec spec) {
         return builder.build();
     }
 
+    public static Optional<Class<?>> findExtensionClass(ClassName clientInterfaceName, Class<?> clazz) {
```

Comment:
Can `clientInterfaceName` be inferred from `clazz`?

Reply:
No, the clazz is determining how the class loader works. I meant to rename it to make it clearer lm do that.